src/main/kotlin/g0401_0500/s0434_number_of_segments_in_a_string/Solution.kt

```diff
@@ -0,0 +1,21 @@
+package g0401_0500.s0434_number_of_segments_in_a_string
+
+// #Easy #String #2022_12_22_Time_167_ms_(80.00%)_Space_35.4_MB_(46.67%)
+
+class Solution {
+    fun countSegments(s: String): Int {
+        var s = s
+        s = s.trim { it <= ' ' }
+        if (s.length == 0) {
+            return 0
+        }
+        val splitted = s.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
+        var result = 0
+        for (value in splitted) {
+            if (value.length > 0) {
+                result++
+            }
+        }
+        return result
+    }
```

Comment:
It may be simple kotlin version

```
public class Solution {
    public int countSegments(String s) {
        s = s.trim();
        if (s.length() == 0) {
            return 0;
        }
        String[] splitted = s.split(" ");
        int result = 0;
        for (String value : splitted) {
            if (value.length() > 0) {
                result++;
            }
        }
        return result;
    }
}
```